SU2_CFD/src/driver_structure.cpp

```diff
@@ -2963,39 +3075,55 @@ su2double CDriver::SetVertexVarCoord(unsigned short iMarker, unsigned short iVer
 
 }
 
-CSingleZoneDriver::CSingleZoneDriver(char* confFile,
-                                     unsigned short val_nZone,
+CGeneralDriver::CGeneralDriver(char* confFile, unsigned short val_nZone,
                                      unsigned short val_nDim) : CDriver(confFile,
                                                                         val_nZone,
                                                                         val_nDim) { }
 
-CSingleZoneDriver::~CSingleZoneDriver(void) { }
+CGeneralDriver::~CGeneralDriver(void) { }
+
+void CGeneralDriver::Run() {
 
-void CSingleZoneDriver::Run() {
 
-  /*--- Run an iteration of the physics within this single zone.
-   We assume that the zone of interest is in the ZONE_0 container position. ---*/
+  unsigned short iZone;
   
-  iteration_container[ZONE_0]->Preprocess(output, integration_container, geometry_container,
-                                          solver_container, numerics_container, config_container,
-                                          surface_movement, grid_movement, FFDBox, ZONE_0);
+  /*--- Run a single iteration of a fem problem by looping over all
+   zones and executing the iterations. Note that data transers between zones
+   and other intermediate procedures may be required. ---*/
   
-  iteration_container[ZONE_0]->Iterate(output, integration_container, geometry_container,
-                                       solver_container, numerics_container, config_container,
-                                       surface_movement, grid_movement, FFDBox, ZONE_0);
+  for (iZone = 0; iZone < nZone; iZone++) {
+    
+    iteration_container[iZone]->Preprocess(output, integration_container, geometry_container,
+                                           solver_container, numerics_container, config_container,
+                                           surface_movement, grid_movement, FFDBox, iZone);
+    
+    iteration_container[iZone]->Iterate(output, integration_container, geometry_container,
+                                        solver_container, numerics_container, config_container,
+                                        surface_movement, grid_movement, FFDBox, iZone);
+    
+    iteration_container[iZone]->Update(output, integration_container, geometry_container,
+                                       solver_container, numerics_container, config_container, surface_movement, grid_movement, FFDBox, iZone);
+    
+    iteration_container[iZone]->Monitor();     /*--- Does nothing for now. ---*/
+    
+    iteration_container[iZone]->Output();      /*--- Does nothing for now. ---*/
+    
+    iteration_container[iZone]->Postprocess(); /*--- Does nothing for now. ---*/
+    
+  }
```

Comment:
Preprocess() and Iterate() are OK here, but I don't understand why you put iteration_container->Update() into the Run() function, considering that the CGeneralDriver::Update() function is also called after the Run() in CDriver::StartSolver(). Basically, it seems like you are calling twice the update of the solver (only for CGeneralDriver).

Also, this seems to correspond to an "old" implementation because I think Monitor(), Output() and Postprocess() are not used at the iteration level anymore (maybe I should have erased them directly since they are apparently still defined empty). They are used at the driver level now.